src/transactions/v2/blockchain_txn_transfer_hotspot_v2.erl

```
@@ -144,10 +148,13 @@ is_valid_owner(
 -spec is_valid(txn_transfer_hotspot_v2(), blockchain:blockchain()) -> ok | {error, any()}.
 is_valid(Txn, Chain) ->
     Ledger = blockchain:ledger(Chain),
-
+    BaseChecks = base_validity_checks(Txn, Ledger, Chain),
     case blockchain:config(?transaction_validity_version, Ledger) of
+        {ok, 3} ->
+            OwnerFeeCheck = {fun() -> owner_can_pay_fee(Txn, Ledger) end, {error, gateway_owner_cannot_pay_fee}},
+            blockchain_utils:fold_condition_checks(BaseChecks ++ [OwnerFeeCheck]);
```

do we fail faster here if we run the `OwnerFeeCheck` first in that list instead of at the end?

---

It's fine, the order is arguable but I think the ownership, nonce, gateway_on_chain checks take precedence anyway.